gemini/use-cases/transcription-summarization/diarized_transcription_and_summarization.ipynb

```diff
@@ -0,0 +1,931 @@
+      "source": [
+        "### Step 1. Audio chunking.\n",
+        "In this step we chunk the audio into smaller chunks of predefined durations, in order to allow transcriptions of large audio files ( lasting several hours) which go beyond the size of the output window of the Gemini 1.5.  (which was 8192 at the time of publication)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "3091633fde85"
+      },
+      "outputs": [],
+      "source": [
+        "storage_client = storage.Client()\n",
+        "\n",
+        "\n",
+        "def chunk_audio(\n",
+        "    audio_url: str,\n",
+        "    output_bucket: str,\n",
+        "    output_format: str = \"wav\",\n",
+        "    chunk_length: int = 600000,\n",
+        "):\n",
+        "    \"\"\"Splits a audio file in GCS into chunks and stores them back in GCS.\n",
+        "\n",
+        "    Args:\n",
+        "        audio_url (str): the GCS url of the audio file to be chunked.\n",
+        "        output_bucket (str):the GCS bucket the chunked audio files would be stored.\n",
+        "        chunk_length_ms (int): Desired chunk length in milliseconds (default: 10\n",
+        "          minutes).\n",
+        "        output_format (str): Output audio format (default: wav).\n",
+        "    \"\"\"\n",
+        "    bucket_name = audio_url.split(\"/\")[2]\n",
+        "\n",
+        "    input_blob_name = \"/\".join(audio_url.split(\"/\")[3:])\n",
+        "\n",
+        "    bucket = storage_client.bucket(bucket_name)\n",
+        "    blob = bucket.blob(input_blob_name)\n",
+        "\n",
+        "    # Download the FLAC file to a temporary local file\n",
+        "    with open(\"temp_audio.\" + output_format, \"wb\") as temp_file:\n",
+        "        storage_client.download_blob_to_file(blob, temp_file)\n",
+        "\n",
+        "    url_audio_chunks = list()\n",
+        "    # Split the audio using pydub\n",
+        "    audio = AudioSegment.from_file(\"temp_audio.\" + output_format, format=output_format)\n",
+        "    chunks = make_chunks(audio, chunk_length)\n",
+        "\n",
+        "    bucket = storage_client.bucket(output_bucket)\n",
+        "    for i, chunk in enumerate(chunks):\n",
+        "        chunk_name_local = (\n",
+        "            f\"{input_blob_name.split('/')[-1].split('.')[0]}_part{i+1}.{output_format}\"\n",
+        "        )\n",
+        "        chunk.export(chunk_name_local, format=output_format)\n",
+        "\n",
+        "        chunk_name_gcs = f\"{input_blob_name.split('.')[0]}_part{i+1}.{output_format}\"\n",
+        "\n",
+        "        # Upload the split chunk back to GCS\n",
+        "        blob = bucket.blob(chunk_name_gcs)\n",
+        "        blob.upload_from_filename(chunk_name_local)\n",
+        "        url_audio_chunks.append(f\"gs://{output_bucket}/{chunk_name_gcs}\")\n",
+        "\n",
+        "    return url_audio_chunks"
```

Comment:
You should be able to simplify some of this code by using `Blob.from_string()`

Something like:

```py
from google.cloud import storage

gcs_uri = "gs://bucket/object"
blob = storage.Blob.from_string(gcs_uri, storage_client)
```

This way you don't have to split and rejoin the paths

Reply:
hi holt, good idea for next time, however for this time since i want to give the user the ability to replace the output bucket with a different one than for the input one: lets keep the existing code if its ok for now